docs/TOOLS.md

```diff
@@ -1,44 +1,178 @@
-2. Unzip the template into your exercises directory, for example `C:\src\exercises`
-2. Install the [Exercism CLI](http://exercism.io/cli)
-3. Open a command prompt to your exercise directory
-4. Add your API key to exercism `exercism configure --key=YOUR_API_KEY`
-5. Configure your source directory in exercism `exercism configure --dir=C:\src\exercises`
-6. [Fetch your first exercise](http://exercism.io/languages/python) `exercism fetch python hello-world`
-7. Open the Exercism solution in Visual Studio
-8. Expand the Exercism.python project
-9. Click on **Show All Files** in Solution Explorer (See below)
-10. The exercise you just fetched will appear greyed out. Right click on the folder and **Include In Project**
-11. Get coding...
+- [Editors and IDEs](#visual-studio-code)
+  - [VS Code](#visual-studio-code)
+    - [Selecting an interpreter](#selecting-the-interpreter)
+    - [Other features](#other-features-1)
+  - [Pycharm](#pycharm)
+    - [Selecting an interpreter](#selecting-the-interpreter-2)
+    - [Other features](#other-features-2)
+  - [Spyder](#spyder-ide)
+    - [Selecting an interpreter](#selecting-the-interpreter-3)
+    - [Other features](#other-features-3)
 
-![Add files](https://raw.githubusercontent.com/exercism/python/main/docs/img/AddFiles.png)
 
-To run the tests, you can do so at the command line, or within Visual Studio.
 
-![Test Explorer](https://raw.githubusercontent.com/exercism/python/main/docs/img/TestExplorer.png)
+---
 
-## Code Style and Linting
+## Environments
 
-There's a style guide called [PEP8](http://legacy.python.org/dev/peps/pep-0008/) that many Python projects adhere to.
-Read it when you get a chance!
+Python environments are like separate Python installations, they can cleanup your workflow and projects by separating the packages you install. Making sure that you don't get bugs generated by something that you imported in another project. 
 
-If you just want a quick overview of some problems in your code, use [pylint](http://www.pylint.org/)!
-It can be pretty picky though, so take its results with a grain of salt.
-If you don't agree with one of its points, that's a good topic for a discussion in the comments for your program!
+There are two major *virtual environments* in use today, `virtualenv` and `conda`. Both of which are generally easy to install.
 
-If you'd rather have a tool take care of your style issues, take a look at [autopep8](https://github.com/hhatto/autopep8)!
-Run `autopep8 -d mycode.py` to get a diff of the changes it proposes and `autopep8 -i mycode.py` to format the code inplace!
+### Venv
+
+Also known as `virtualenv`, *venv* is a light-weight solution to virtual environments. It creates a separate Python binary for every virtual environment and stores that inside your directory.
+
+Installing `venv` is easy using `pip`:
+
+```bash
+$ python3 -m pip install virtualenv
+Successfully installed virtualenv-20.8.1
+```
```

Comment:
`venv` is now part of the standard lib:  https://docs.python.org/3/library/venv.html, so you no longer have to install `virtualenv`, you can just invoke it.  😄     Also ... Real Python has a good article on `venv` for reference.  https://realpython.com/python-virtual-environments-a-primer/.

Reply:
[`virtualenvironmentwrapper`](https://virtualenvwrapper.readthedocs.io/en/latest/) is  very handy for management, but might also be a bit much for anyone just starting with envs.

Reply:
I'll remove the installing `venv` part, then! 

I am worried that realpython.com might be unusable for some students, since they now allow up to a max amount of unique visits a day. I don't think that people should see it as us trying to make them buy a realpython membership if they get hit by a paywall.

I'll add a small piece about `virtualenvironmentwrapper` at the end of the `venv` part, a bit like the ones called `other features` at the end of an IDE or Editor docs.

Reply:
Totally agree on the RealPython thing, btw.  I am (selfishly) sad that they are metering and paywalling now.